src/metrics/mod.rs

```diff
@@ -78,6 +80,22 @@ pub fn init() -> crate::Result<()> {
     // case it doesn't _do_ much other than shepherd into the registry and
     // update the cardinality counter, see above, as needed.
     let recorder = VectorRecorder::new(registry);
+
+    #[cfg(any(feature = "sources-kafka", feature = "sinks-kafka"))]
+    let recorder = {
+        let patterns = vec![
+            "kafka_requests_total",
+            "kafka_requests_bytes_total",
+            "kafka_responses_total",
+            "kafka_responses_bytes_total",
+            "kafka_produced_messages_total",
+            "kafka_produced_messages_bytes_total",
+            "kafka_consumed_messages_total",
+            "kafka_consumed_messages_bytes_total",
+        ];
+        AbsoluteLayer::from_patterns(patterns).layer(recorder)
```

Comment:
Have you considered throwing in a `use_dfa` here? The docs for metrics-util claims it matches 2-4 times as fast. https://docs.rs/metrics-util/0.6.2/metrics_util/layers/struct.AbsoluteLayer.html#method.use_dfa

Reply:
That's a good catch, but unfortunately it is already enabled:

> Defaults to `true`.

https://docs.rs/metrics-util/0.6.2/src/metrics_util/layers/absolute.rs.html#111.